src/subclass/cell_renderer.rs

```diff
@@ -0,0 +1,658 @@
+    let imp = instance.get_impl();
+    let wrap: CellRenderer = from_glib_borrow(ptr);
+    let widget: Widget = from_glib_borrow(wdgtptr);
+
+    let (min_size, nat_size) = imp.get_preferred_width(&wrap, &widget);
+    if !minptr.is_null() {
+        *minptr = min_size;
+    }
+    if !natptr.is_null() {
+        *natptr = nat_size;
+    }
+}
+
+unsafe extern "C" fn cell_renderer_get_preferred_height_for_width<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkCellRenderer,
+    wdgtptr: *mut gtk_sys::GtkWidget,
+    width: c_int,
+    min_height_ptr: *mut c_int,
+    nat_height_ptr: *mut c_int,
+) where
+    T: CellRendererImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: CellRenderer = from_glib_borrow(ptr);
+    let widget: Widget = from_glib_borrow(wdgtptr);
+
+    let (min_height, nat_height) = imp.get_preferred_height_for_width(&wrap, &widget, width);
+    if !min_height_ptr.is_null() {
+        *min_height_ptr = min_height;
+    }
+    if !nat_height_ptr.is_null() {
+        *nat_height_ptr = nat_height;
+    }
+}
+
+unsafe extern "C" fn cell_renderer_get_preferred_height<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkCellRenderer,
+    wdgtptr: *mut gtk_sys::GtkWidget,
+    minptr: *mut c_int,
+    natptr: *mut c_int,
+) where
+    T: CellRendererImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: CellRenderer = from_glib_borrow(ptr);
+    let widget: Widget = from_glib_borrow(wdgtptr);
+
+    let (min_size, nat_size) = imp.get_preferred_height(&wrap, &widget);
+    if !minptr.is_null() {
+        *minptr = min_size;
+    }
+    if !natptr.is_null() {
+        *natptr = nat_size;
+    }
+}
+
+unsafe extern "C" fn cell_renderer_get_preferred_width_for_height<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkCellRenderer,
+    wdgtptr: *mut gtk_sys::GtkWidget,
+    height: c_int,
+    min_width_ptr: *mut c_int,
+    nat_width_ptr: *mut c_int,
+) where
+    T: CellRendererImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: CellRenderer = from_glib_borrow(ptr);
+    let widget: Widget = from_glib_borrow(wdgtptr);
+
+    let (min_width, nat_width) = imp.get_preferred_width_for_height(&wrap, &widget, height);
+    if !min_width_ptr.is_null() {
+        *min_width_ptr = min_width;
+    }
+    if !nat_width_ptr.is_null() {
+        *nat_width_ptr = nat_width;
+    }
+}
+
+unsafe extern "C" fn cell_renderer_get_aligned_area<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkCellRenderer,
+    wdgtptr: *mut gtk_sys::GtkWidget,
+    flags: gtk_sys::GtkCellRendererState,
+    cellarea: *const gdk_sys::GdkRectangle,
+    alignedptr: *mut gdk_sys::GdkRectangle,
+) where
+    T: CellRendererImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: CellRenderer = from_glib_borrow(ptr);
+    let widget: Widget = from_glib_borrow(wdgtptr);
```

Comment:
All these type annotations should not be needed but feel free to keep them

Reply:
ok